### PR TITLE
tech: Add Tooltip UI component

### PR DESCRIPTION
**Pull request by Youakeem**
## Describe your changes
Add a composable version of the Tooltip component in the `ui` package

## Justify why they are needed
- Works with composed components
- Needed for components in the `ui` package

## Checklist before requesting a review

- [x] I have performed a self-review of my code

### CODE REVIEW

**Comment by Youakeem** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4708?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4710** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4710?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4709** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4709?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4708** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4708?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4707** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4707?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4706** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4706?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4705** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4705?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4704** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4704?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4703** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4703?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Youakeem and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/ui/src/components/Tooltip.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  '0%': { opacity: 0, transform: 'translateY(10px)' },
+  '100%': { opacity: 1, transform: 'translateY(0)' },
+})
+
+const slideDownAndFadeAnimation = keyframes({
+  '0%': { opacity: 0, transform: 'translateY(-10px)' },
+  '100%': { opacity: 1, transform: 'translateY(0)' },
+})
+
+const Content = styled(TooltipPrimitive.Content)({
```
Comment (alebedev) [inline packages/ui/src/components/Tooltip.tsx:53]: Is there a reason to use emotion styles in new component?

File: packages/ui/src/components/Tooltip.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
+
```
Comment (alebedev) [inline packages/ui/src/components/Tooltip.tsx:1]: This PR adds component, but does not use it. How do we test that it works as intended? We have few places where radix tooltip was used directly, can we replace one of them so these changes become testable?

File: packages/ui/src/components/Tooltip.tsx (hunk 1 above)
Comment (Youakeem) [reply packages/ui/src/components/Tooltip.tsx:53]: Nope, I just copied it as is. We can come back to it and rewrite it. I just needed the component to move forward with the rest of the work.

File: packages/ui/src/components/Tooltip.tsx (hunk 2 above)
Comment (Youakeem) [reply packages/ui/src/components/Tooltip.tsx:1]: It's used in the next PR. But you've got a point. I should add some storybook stories.

**Review by alebedev — APPROVED** [review]

**Review by guilhermespopolin — APPROVED** [review]

**Comment by Youakeem** [comment]
### Merge activity

* **Sep 19, 3:16 AM EDT**: @Youakeem started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4708).

**Pull request merged**